### ISSUE TITLE
Problem with Drupal Core  

### ISSUE THREAD

**Issue by vacho** [christophegalli/entity_d8#4]
I try install this module today April 9, 2014 and I have this error

Fatal error: Declaration of Drupal\foo_bar\Entity\FooBar::preCreate() must be compatible with Drupal\Core\Entity\EntityInterface::preCreate(Drupal\Core\Entity\EntityStorageInterface $storage, array &$values) in /var/www/devd8/modules/entity_d8/lib/Drupal/foo_bar/Entity/FooBar.php on line 46

**Comment by qdelance** [comment]
Which D8 version are you using ?
Was working fine on latest alpha AFAIR.
I guess you're using latest dev version ? It has a changing API. In that case you have to go back to core source to get the new signature and update source code accordingly.

@christophegalli : what are your plans for this module ? do you want to wait for next alpha to update code updates ? or is it possible to submit PR to keep up with upstream changes ?

**Comment by christophegalli** [comment]
I always work on the latest dev pull. The content_entity_example is kept up to date. this specific repo here will be adapted to the latest dev from time to time. it is my private repo that I showed you to visualize hook_views_data....

btw, the necessary corrections are minimal. you will find them all in the content_entity_example patch.

**Comment by christophegalli** [comment]
@qdelance : the development is mostly happening on the content_entity_example which may eventually end up in the D8 release. As you know, it is a bit of a cat and mouse game as long as API's of D8 still change. 

The documentation https://drupal.org/node/2192175 will eventually be adapted to the example module.

The next planned step is to add a 'real world' example with bundle, views integration and all the other options. This will live within the exapmle module.

What exactly do you mean by PR?

**Comment by qdelance** [comment]
Thx for this info. I missed the fact that most of the work was happening in content_entity_example (master branch is empty, I see the commits now :), so I'll keep following this project now : thanks for your time, this kind of module really helps keeping up with API changes (hard to follow if you're not a core dev ;)

PR = pull request, I try to submit patches to D8 tutorials/test projects I find on the web to help authors being in sync with latest dev. I seems it is not necessary in this case !
